text/0000-build-std/5-stage-1b.md

```diff
@@ -0,0 +1,986 @@
+
+↩ [*Features*][features]
+
+## Why default to public for the implicit standard library dependencies?
+[rationale-implicit-public]: #why-default-to-public-for-the-implicit-standard-library-dependencies
+
+There are crates building on stable which re-export from the standard library.
+If the implicit standard library dependency were not public then these crates
+would start to trigger the `exported_private_dependencies` lint when upgrading
+to a version of Cargo with an implicit standard library dependency.
+
+↩ [*Public and private dependencies*][public-and-private-dependencies]
+
+## Why follow the default privacy of explicit standard library dependencies?
+[rationale-explicit-private]: #why-follow-the-default-privacy-of-explicit-standard-library-dependencies
+
+This may be unintuitive when a user first writes an explicit standard library
+dependency, triggering the `exported_private_dependency` lint, but this would be
+caught immediately by the user. However, it is also unintuitive that the default
+for privacy of a explicitly written dependency would depend on which crate the
+dependency was (i.e. the standard library has a different default than
+everything else).
+
+↩ [*Public and private dependencies*][public-and-private-dependencies]
+
+## Why not support explicit standard library dependencies in `build-dependencies`?
+[rationale-no-deps-in-build-deps]: #why-not-support-explicit-standard-library-dependencies-in-build-dependencies
+
+`build-dependencies` only apply to build scripts which are run on the host
+toolchain. There is little advantage to using a custom standard library with
+build scripts as they are not part of the final output artifact and anywhere
+they can run already has a toolchain with host tools and a pre-built standard
+library.
+
+↩ [*`dev-dependencies` and `build-dependencies`*][dev-dependencies-and-build-dependencies]
+
+## Why add standard library crates to Cargo's index?
+[rationale-cargo-index]: #why-add-standard-library-crates-to-cargos-index
+
+When Cargo builds the dependency graph, it is driven by the index (not
+`Cargo.toml`), so builtin dependencies need to be included in the index.
+
+↩ [*Registries*][registries]
+
+## Why add a new key to Cargo's registry index JSON schema?
+[rationale-cargo-builtindeps]: #why-add-a-new-key-to-cargos-registry-index-json-schema
+
+Cargo's [registry index schema][cargo-json-schema] is versioned and making a
+behaviour-of-Cargo-modifying change to the existing `deps` keys would be a
+breaking change. Each package is published under one particular version of the
+schema, meaning that older versions of Cargo cannot use newer versions of
+packages which are defined using a schema it does not have knowledge of.
+
+Cargo ignores packages published under an unsupported schema version, so older
+versions of Cargo cannot use newer versions of packages relying on these
+features. New schema versions are disruptive to users on older toolchains and
+should be avoided where possible.
```

Comment:
>  so older
versions of Cargo cannot use newer versions of packages relying on these
features

Thats true anyways because the `Cargo.toml` won't be parseable.

Extensions to an existing schema version that can't be understood by older cargos is disruptive in terms of the resolver acts as if the version just doesn't exist.  There may still be cases where the error reporting is sub-par but a lot of the problems were fixed as of 1.85 (https://github.com/rust-lang/cargo/pull/14927).  The further we are from 1.85, the less of an issue this is.

As for bumping schema versions, I vaguely remember that being a lot more disruptive.  I think its related to alternative registries.  That said, the error reporting work I did in 1.85 also applies here.

Reply:
We'll make sure to include this context.